### ISSUE TITLE
ranging from X to Y

### ISSUE THREAD

**Issue by arademaker** [UniversalDependencies/UD_English-EWT#266]
```
email-enronsent22_01-0017> So we now have three grandsons and three granddaughters ranging from about 2 years - 4 months down to a few days.

─┮  
 │ ╭─╼ So ADV advmod 1 4  
 │ ├─╼ we PRON nsubj 2 4  
 │ ├─╼ now ADV advmod 3 4  
 ╰─┾ have VERB root 4 0  
   │ ╭─╼ three NUM nummod 5 6  
   ├─┾ grandsons NOUN obj 6 4  
   │ │ ╭─╼ and CCONJ cc 7 9  
   │ │ ├─╼ three NUM nummod 8 9  
   │ ├─┶ granddaughters NOUN conj 9 6  
   │ ╰─┮ ranging VERB acl 10 6  
   │   │ ╭─╼ from ADP case 11 14  
   │   │ │ ╭─╼ about ADV advmod 12 13  
   │   │ ├─┶ 2 NUM nummod 13 14  
   │   ╰─┾ years NOUN obl 14 10  
   │     │ ╭─╼ - SYM cc 15 17  
   │     │ ├─╼ 4 NUM nummod 16 17  
   │     ├─┶ months NOUN conj 17 14  
   │     │ ╭─╼ down ADV advmod 18 22  
   │     │ ├─╼ to ADP case 19 22  
   │     │ ├─╼ a DET det 20 22  
   │     │ ├─╼ few ADJ amod 21 22  
   │     ╰─┶ days NOUN nmod 22 14  
   ╰─╼ . PUNCT punct 23 4  
```

This analysis of `nmod(years, days)` seems wrong to me, am I right? How should we analyze this kind of structure?

**Comment by nschneid** [comment]
Yes I think for now, it should be two PPs attaching to the verb. (There is perhaps an argument to be made that "from X to Y" is a special construction reminiscent of coordination, but that is certainly not the standard practice currently.)

**Comment by nschneid** [comment]
[instances of "from X" + "to Y" where Y is a dependent of X](http://match.grew.fr/?corpus=UD_English-EWT@dev&custom=6195986786da8)

**Comment by sylvainkahane** [comment]
In some French corpora we analyzed "de X à Y" as a coordination :
http://match.grew.fr/?corpus=SUD_French-Rhapsodie@latest&custom=61961d959f3e2

They are good reasons to do that in French.

1) "de X à Y" can occupy different positions, including the determiner position:

> _on attend de 10 à 20 personnes_ 'we expect 10 to 20 people' (_de_ is optional)

2) the order is fixed (French as a freer order than English)

_le train de Paris à Londres_ 'the train from Paris to London'
_*le train à Londres de Paris_ 'the train from Paris to London'

3) NCC is possible:

_un vent de force 4 le matin à force 5 l'après-midi_ 'a wind from force 4 in the morning to force 5 in the afternoon'

4) it can be cleft (normally only one constituent can)

> _c'est de chez toi à chez moi que c'est le plus court 'it's from your place to mine that it's the shortest'

**Comment by nschneid** [comment]
> In some French corpora we analyzed "de X à Y" as a coordination : http://match.grew.fr/?corpus=SUD_French-Rhapsodie@latest&custom=61961d959f3e2

That is SUD. Would the UD equivalent for "aller de la place Notre-Dame à La Nef Chavant" be:

    obl(aller, place)
    case(place, de)
    conj(place, Nef)
    case(Nef, à)

or perhaps

    obl(aller, place)
    cc:preconj(place, de)
    conj(place, Nef)
    cc(Nef, à)

?

English equivalents of the above examples:

1. We expect (from) 10 to 20 people.
   * Cf. We expect 10 or 20 people. (interpreted loosely as "about 10 or 20")
   * Cf. We expect over/up to/at least 10 people [approximator construction]. Notably, "from" and "to" don't work by themselves in this construction:
      - \*We expect [to 20] people.
      - \*We expect [from 20] people.
2. _The train to London from Paris_ is fine, but not \*_We expect to 20 from 10 people_.
3. _Winds ranged from [20mph this morning] to [40mph this afternoon]._
4. Having trouble thinking of an *it*-cleft example in English.

It's worth considering "between X and Y" as well—I suspect that would have a similar analysis, with the differences that (i) "between" is mandatory (\*_We expect 10 and 20 people_); (ii) "and" is readily recognizable as a coordinating conjunction, whereas "to" is *mainly* a preposition (arguably a conjunction when expressing a range, though).

**Comment by amir-zeldes** [comment]
I think wanting to connect the "from" phrase to the "to" phrase in some contexts (e.g. when there is no verb) could make sense, but I would still expect the normal PP deprels (nmod/obl etc.), and I imagine most users of the data would expect those as well. 

We just ran into a similar issue in the upcoming GUM8 data, where a present is described as "From Santa, to Dan" (no other words in the sentence), where I think the simplest option is obl (treating "from Santa as root", and "to Dan" as a modifier of the predication, not of "Santa"). Alternatives we considered were orphan with a reconstructed verb (seems excessive), nmod (seems wrong since it's not really adnominal) and parataxis.

**Comment by sylvainkahane** [comment]
Our UD analysis for "aller de la place Notre-Dame à La Nef Chavant" is:

> obl(aller, place)
> case(place, de)
> conj(place, Nef)
> case(Nef, à)

In this example, the two complements can be inverted: "aller à La Nef Chavant de la place Notre-Dame".
In the other example, with a figurative sense of ALLER 'go', the inversion is impossible:

> ça va vraiment de romans étrangers à quelques romans français contemporains
> 'it really goes/ranges from foreign novels to some contemporary French novels'
> *ça va vraiment à quelques romans français contemporains de romans étrangers 

For the POS, _de_ 'from' and _à_ 'to' are ADPs. It would be strange to analyze them as CCONJs, even if they play a role in the coordination. These prepositions keep their original meaning (origin vs destination) in this construction. Same thing in English.

**Comment by nschneid** [comment]
> We just ran into a similar issue in the upcoming GUM8 data, where a present is described as "From Santa, to Dan" (no other words in the sentence), where I think the simplest option is obl (treating "from Santa as root", and "to Dan" as a modifier of the predication, not of "Santa"). Alternatives we considered were orphan with a reconstructed verb (seems excessive), nmod (seems wrong since it's not really adnominal) and parataxis.

"From Santa, to Dan" or "To Dan, from Santa" is not the "from X to Y" construction we're talking about. It's perhaps a convention of the genre to put these two PPs next to each other—I would call it parataxis or asyndetic coordination (could be paraphrased with "and").

**Comment by amir-zeldes** [comment]
Yeah, I realize it's not the same, I just thought it was similar in that they seem to belong together, though semantically there is an implied predicate whose valency they fill, and we have to connect them somehow due to the environment. I'd still be against calling it a coordination, that's something different again (in a coordination I'd expect both conjuncts to saturate the same single role as a coordinate phrase, and in "I went from La Nef Chavant to Notre Dame" they saturate different roles - source and destination)

**Comment by arademaker** [comment]
Thank you @nschneid , so maybe we can keep this issue open since your query return many inconsistent analysis, right? One situation where we don't have a common VERB head is 

> The 22 member **countries** of the Arab league, from **Mauritania** to **the Gulf States**, have a total population of 300 millions, larger than the US and almost as large as the EU before its expansion.

**Comment by arademaker** [comment]
My question was motivated by the Ontonotes example:

> Through a series of transitional forms for state capitalism , **ranging from low to high levels** , and after taking into consideration China 's national conditions , the Party implemented " peaceful redemption " on capitalist owned means of production based on the visions of Marx and Lenin .

**Comment by nschneid** [comment]
UniversalDependencies/docs#871
